src/notify/notify.c

```diff
@@ -203,7 +203,7 @@ static int run(int argc, char* argv[]) {
                 return r;
 
         if (arg_booted)
-                return sd_booted() <= 0;
+                return -(sd_booted() <= 0);
```

Comment:
`run` only returns 0 or negative apart from this place. Please instead change DEFINE_MAIN_FUNCTION to DEFINE_MAIN_FUNCTION_WITH_POSITIVE_FAILURE below. And then propagate the error here (possibly with a log_debug line), and return the inverted value otherwise.

Reply:
I saw that there is `DEFINE_MAIN_FUNCTION_WITH_POSITIVE_FAILURE`, but there are a lot more returns in that function and I wasn't sure about them. I'll change it then.